src/js/state.js

```diff
@@ -41,6 +41,14 @@ export default async function () {
   } else {
     ciphers = ciphers;
   }
+  if (form['config'].value === 'old') {
+    if (configs[server].usesOpenssl !== false) {
+      // set SECLEVEL=0 for openssl 3 to support TLSv1 and TLSv1.1 for Old config
+      if (minver('3.0.0', form['openssl'].value) === true) {
+        ciphers.unshift('@SECLEVEL=0');
```

Comment:
Haven't looked into properly, only quick observation: https://gst-os3--mozsslconf-dev.netlify.app/#server=oraclehttp&version=12.2.1&config=old&openssl=3.3.2&guideline=5.7

```nginx
# old configuration
SSLProtocol             All 
SSLCipherSuite          @SECLEVEL=0:@SECLEVEL=0:@SECLEVEL=0:@SECLEVEL=0:@SECLEVEL=0:@SECLEVEL=0:@SECLEVEL=0:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-
```

Not that it wouldn't work, right? (I haven't tried tbh;D) but it just doesn't look right.

Reply:
That is funny.  I believe I had tested with stunnel.hbs, and hadn't looked across the rest.   I'll see about modifying the javascript tomorrow or later in the weekend.

Reply:
I pushed a change to fix this.  If you're concerned with the condition `if (ciphers[0] !== '@SECLEVEL=0') {` on a possibly empty ciphers list (which should not happen for the Old config), then I can change the logic to something similar used to set `usesDhe` further below in the same routine: `if (ciphers.join(":").includes("@SECLEVEL=0") === false) {`.

Reply:
Not an urgent concern of mine, but the repetition that I now have to check for suggests that there are some performance gains on the client-side to be had by eliminating unnecessary repetition of work.  Obviously, things need to be re-evaluated, when a config change is made by the user, but it should not need to be re-evaluated 5 times.

Reply:
I was basically more concerned about _how_ it got iterated over so many times in the first place;)

> I was trying to avoid creating a new variable for @SECLEVEL=0 that had to be pasted onto the cipher list and have code added to each and every hbs

Sure, that is similar to tls13 logic also filtering out stuff in state.js before passing it on opaquely to the template output. That's definitely a good place!

(BTW, it's plain JS here, not HBS, so you don't have to nest the conditions so bizarrely (:) and can use falsy bool values without too much remorse really…)

Reply:
I am not a javascript developer, so I appreciate any recommendations for better coding idioms or coding standard preferences.

Please add to your wishlist an issue to ask why this code is being run repeatedly for some templates.  It did not appear to run repeatedly for the (new) stunnel template.